library/include/G4CMPSolidUtils.hh

```diff
@@ -0,0 +1,172 @@
+/***********************************************************************\
+ * This software is licensed under the terms of the GNU General Public *
+ * License version 3 or later. See G4CMP/LICENSE for the full license. *
+\***********************************************************************/
+
+/// \file library/include/G4CMPSolidUtils.hh
+/// \brief This class contains custom utilities for G4VSolids that are not
+/// directly included in the G4VSolid class. Some example of utilites are:
+/// 1) Adjusting a position to the nearest surface point without the surface normal.
+/// 2) Adjusting a particle to the nearest edge along the surface.
+///
+/// The user will need to create this class directly with a G4VSolid and/or
+/// touchable, or can initialize these objects later on.
+///
+/// Note: All coordinates/vectors passed into functions must be in the solid's local coordinate system.
+//
+// $Id$
+//
+// 20250424  G4CMP-465 -- Create G4CMPSolidUtils class.
+
+#ifndef G4CMPSolidUtils_hh
+#define G4CMPSolidUtils_hh 1
+
+#include "globals.hh"
+#include "G4AffineTransform.hh"
+#include "G4RotationMatrix.hh"
+#include "G4ThreeVector.hh"
+
+class G4VSolid;
+class G4VTouchable;
+
+class G4CMPSolidUtils {
+  public:
+    // Default constructor
+    G4CMPSolidUtils() : theSolid(0), theTransform(G4AffineTransform()), verboseLevel(0),
+                        verboseLabel("G4CMPSolidUtils") {;}
+
+    // Direct constructor with solid & transform for client code in local frame
+    G4CMPSolidUtils(const G4VSolid* solid, G4int verbose, G4String vLabel);
```

Comment:
I didn't catch this earlier, sorry!  Use `const G4String& vLabel` for the constructor and Set function argument.  Otherwise, you end up creating and deleting a temporary.